toothpick-compiler/src/test/java/toothpick/compiler/factory/FactoryRegistryTest.java

```diff
@@ -2,254 +2,459 @@
+                "import java.util.Map;", //
+                "import toothpick.registries.factory.AbstractFactoryRegistry;", //
+                "", //
+                "public final class FactoryRegistry extends AbstractFactoryRegistry {", //
+                "  private final Map<String, Integer> classNameToIndex = new HashMap<>();", //
+                "", //
+                "  public FactoryRegistry() {", //
+                "    registerGroup0();", //
+                "    registerGroup1();", //
+                "  }", //
+                "", //
+                "  private void registerGroup0() {", //
+                "    classNameToIndex.put(\"test.Class0\", 0);", //
+                "    classNameToIndex.put(\"test.Class1\", 1);", //
+                "    classNameToIndex.put(\"test.Class2\", 2);", //
+                "  }", //
+                "", //
+                "  private void registerGroup1() {", //
+                "    classNameToIndex.put(\"test.Class3\", 3);", //
+                "    classNameToIndex.put(\"test.Class4\", 4);", //
+                "  }", //
+                "", //
+                "  @Override", //
+                "  public <T> Factory<T> getFactory(Class<T> clazz) {", //
+                "    Factory<T> factory = getFromThisRegistry(clazz);", //
+                "    if (factory == null) {", //
+                "      return getFactoryInChildrenRegistries(clazz);", //
+                "    }", //
+                "    return factory;", //
+                "  }", //
+                "", //
+                "  private <T> Factory<T> getFromThisRegistry(Class<T> clazz) {", //
+                "    Integer index = classNameToIndex.get(clazz.getName());", //
+                "    if (index == null) {", //
+                "      return null;", //
+                "    }", //
+                "    int groupIndex = index / 3;", //
+                "    switch(groupIndex) {", //
+                "      case 0: return getFromGroup0(index);", //
+                "      case 1: return getFromGroup1(index);", //
+                "    }", //
+                "    return null;", //
+                "  }", //
+                "", //
+                "  private <T> Factory<T> getFromGroup0(int index) {", //
+                "    switch(index) {", //
+                "      case 0: return (Factory<T>) new test.Class0$$Factory();", //
+                "      case 1: return (Factory<T>) new test.Class1$$Factory();", //
+                "      case 2: return (Factory<T>) new test.Class2$$Factory();", //
+                "    }", //
+                "    return null;", //
+                "  }", //
+                "", //
+                "  private <T> Factory<T> getFromGroup1(int index) {", //
+                "    switch(index) {", //
+                "      case 3: return (Factory<T>) new test.Class3$$Factory();", //
+                "      case 4: return (Factory<T>) new test.Class4$$Factory();", //
```

Comment:
wouldn't it be clearer to start the indexes from 0? Index within the Group.

Reply:
I think it's clearer this way: the indices here are exactly the same as in the corresponding map-filling method. And no need for a modulus operation.

Reply:
sounds good